### ISSUE TITLE
Stack overflow when trying to remove an item from a treeview

### ISSUE THREAD

**Issue by ffquintella** [AvaloniaUI/Avalonia#12681]
I get a stack overflow error every time I try to remove and Item from a collection binded to my tree view. I debugged and the problem seems to be inside one of the avalonia classes. 

This is the tree view 

            
            <TreeView Grid.Row="1" ItemsSource="{Binding Nodes, Mode=TwoWay}"
                      SelectedItem="{Binding SelectedNode, Mode=TwoWay}">
                <TreeView.ItemTemplate>
                    <TreeDataTemplate ItemsSource="{Binding SubNodes, Mode=TwoWay}" >
                        <TextBlock Text="{Binding Title}" IsVisible="{Binding IsVisible, Mode=TwoWay}"/>
                    </TreeDataTemplate>
                </TreeView.ItemTemplate>
            </TreeView>

This is the treenode class:
public class TreeNode: ReactiveObject
{
    
    private ObservableCollection<TreeNode>? _subNodes;

    public ObservableCollection<TreeNode>? SubNodes
    {
        get => _subNodes;
        set
        {
            if (value == null) return;
            this.RaiseAndSetIfChanged(ref _subNodes, value);
        }
    }
    
    public string Title { get; }
    
    public int EntityId { get; }

    public bool IsVisible { get; set; } = true;

    public TreeNode(string title)
    {
        Title = title;
    }

    public TreeNode(string title, int entityId, ObservableCollection<TreeNode> subNodes)
    {
        EntityId = entityId;
        Title = title;
        SubNodes = subNodes;
    }
}

And this is the instruction that starts the problem:

Nodes!.Remove(rootNode);

**To Reproduce**
I guess it would happen with a treeview with similar structure while trying to remove any item.

I am using avalonia 11.0.4

**Comment by IanRawley** [comment]
Is there a reason that your ItemsSource and the TextBlock's IsVisible bindings are TwoWay?  I'm not sure how that would cause a StackOverflow exception rather than an NRE, but it does look strange to me.

**Comment by timunie** [comment]
Yeah, use two-way binding only if you know you need to write back things that the user set in the view to the ViewModel. Example:
```
TextBox.Text
CheckBox.IsChecked
ToggleButton.IsChecked
```

but the user cannot change `IsVisible`, `ItemsSource`, ... -> so these properties are one-way bindings. Only the ViewModel can change it and inform the View.

Aynway, I tried  your provided sample and I'm unable to reproduce. So please upload a very minimal App to reproduce. Thanks 🙏

**Comment by timunie** [comment]
@ffquintella I'm moving this into Q&A because I think there is something going on in your ViewModel (as I can't reproduce). If we get your minimal sample and it is an Avalonia bug, we will open a new issue with detailed steps to reproduce, okay?
